### DISCUSSION TITLE
Tracing and Session support on Integrations like LiteLLM Proxy and others and Is LangFuse logging a background process?

### DISCUSSION THREAD

**Discussion by Harryalways317** [langfuse/langfuse#1536] [Support] [answered]
Hey People, there isn't any clear documentation on how to add tracing and tags while using integrations, also will the call to langfuse handled asynchronsly? so the main return wont be blocked,

i am using LiteLLM -> Langfuse -> Guardrails, so this is causing delay as generation is faster but next 2 steps are slow, i know the Guardrails will take time especially when language models are involved, but just wanted to confirm that does langfuse logs as a background process

**Comment by marcklingen** [comment, marked as answer]
> also will the call to langfuse handled asynchronsly? so the main return wont be blocked,

Langfuse works fully async on a separate background thread to send events to the API.

> Hey People, there isn't any clear documentation on how to add tracing and tags while using integrations

The LiteLLM docs are quite extensive on this: https://litellm.vercel.app/docs/observability/langfuse_integration

**Reply by Harryalways317** [reply]
Thanks
